connectors/sources/google.py

```diff
@@ -0,0 +1,304 @@
+            )
+
+        return await anext(self._execute_api_call(resource, method, _call_api, kwargs))
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+    )
+    async def _execute_api_call(self, resource, method, call_api_func, kwargs):
+        """Execute the API call with common try/except logic.
+        Args:
+            resource (aiogoogle.resource.Resource): Resource name for which the API call will be made.
+            method (aiogoogle.resource.Method): Method available for the resource.
+            call_api_func (function): Function to call the API with specific logic.
+            kwargs: Additional arguments for the API call.
+        Raises:
+            exception: An instance of an exception class.
+        Yields:
+            async generator: Response returned by the resource method.
+        """
+        try:
+            async with Aiogoogle(
+                service_account_creds=self.service_account_credentials
+            ) as google_client:
+                workspace_client = await google_client.discover(
+                    api_name=self.api, api_version=self.api_version
+                )
+
+                if isinstance(resource, list):
+                    resource_object = getattr(workspace_client, resource[0])
+                    for nested_resource in resource[1:]:
+                        resource_object = getattr(resource_object, nested_resource)
+                else:
+                    resource_object = getattr(workspace_client, resource)
+                method_object = getattr(resource_object, method)
+
+                async for item in call_api_func(google_client, method_object, kwargs):
+                    yield item
+
+        except AttributeError as exception:
+            self._logger.error(
+                f"Error occurred while generating the resource/method object for an API call. Error: {exception}"
+            )
+            raise
+        except HTTPError as exception:
+            self._logger.warning(
+                f"Response code: {exception.res.status_code} Exception: {exception}."
+            )
+            raise
+        except Exception as exception:
+            self._logger.warning(f"Exception: {exception}.")
+            raise
+
+
+def remove_universe_domain(json_credentials):
+    if "universe_domain" in json_credentials:
+        json_credentials.pop("universe_domain")
+
+
+class GoogleDirectoryClient:
+    def __init__(self, json_credentials, customer_id, timeout=DEFAULT_TIMEOUT):
+        remove_universe_domain(json_credentials)
+        self._customer_id = customer_id
+        self._client = GoogleServiceAccountClient(
+            json_credentials=json_credentials,
+            api=APIS[API.DIRECTORY][IDENTIFIER],
+            api_version=APIS[API.DIRECTORY][VERSION],
+            scopes=[Scope.ADMIN_DIRECTORY_USER_READONLY.value],
```

Comment:
IMO it's overcomplicated and all the constants can be inlined

Reply:
Inlined with [2ea5d7b](https://github.com/elastic/connectors-python/pull/1317/commits/2ea5d7be6115c204c77b90b6caf9770410884a43)